controllers/users/populateUser.js

```diff
@@ -3,7 +3,12 @@ const {sequelize} = require('../../databases/models');
 
 module.exports = async (req, res) => {
   try {
-    const {limit = 50, offset = 0} = req.query;
+    const {limit = 50, offset = 0, allow_anon_dm = ''} = req.query;
+
+    let filterAllowDm = '';
+    if (allow_anon_dm !== '') {
+      filterAllowDm = `AND users.allow_anon_dm = ${allow_anon_dm}`;
+    }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a potential SQL injection vulnerability in the way you're handling the `allow_anon_dm` query parameter. Directly interpolating user-provided values into SQL queries can lead to malicious SQL code being executed. It's better to use parameterized queries or prepared statements to prevent this.

```diff
-    const {limit = 50, offset = 0, allow_anon_dm = ''} = req.query;
+    const {limit = 50, offset = 0, allow_anon_dm = null} = req.query;

-    let filterAllowDm = '';
-    if (allow_anon_dm !== '') {
-      filterAllowDm = `AND users.allow_anon_dm = ${allow_anon_dm}`;
-    }
+    let filterAllowDm = {};
+    if (allow_anon_dm !== null) {
+      filterAllowDm = {allow_anon_dm: allow_anon_dm};
+    }
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->